### PR TITLE
Add resolved rules and comments.

### PR DESCRIPTION
**Pull request by TrevorKarjanis**
- Add rules and comments that have been resolved.

Notes
- The style of the rules definition in index.js will evolve as rules are added.

Linted and tested
- Hakuna matata

### CODE REVIEW

**Comment by rajsite** [comment]
Could we update the PR to reflect the in person discussion for `space-before-function-paren`:
```
'space-before-function-paren': ['error', { anonymous: 'always', named: 'never', asyncArrow: 'always' }]
```

which is reflected by the following [examples](https://eslint.org/docs/rules/space-before-function-paren#anonymous-always-named-never-asyncarrow-always):
```js
function foo() {
    // ...
}

var bar = function () {
    // ...
};

class Foo {
    constructor() {
        // ...
    }
}

var foo = {
    bar() {
        // ...
    }
};

var foo = async (a) => await a
```

**Comment by TrevorKarjanis** [comment]
> Could we update the PR to reflect the in person discussion for `space-before-function-paren`:

@rajsite Yes, no problem.

**Comment by rajsite** [comment]
@TrevorKarjanis looks like there are merge conflicts to resolve since the .eslintrc moved to the root

File: index.js [hunk 1]
```diff
@@ -62,17 +74,68 @@ module.exports = {
...
+
         /*
             Migrating from var is not trivial, so there may be exceptions for older projects.
         */
         'no-var': 'error',
 
+        /*
+            Defining classes and variables before their use can cause errors. However, defining
+            functions and methods before their use is a common programming practice for
+            readability.
```
Comment (rajsite) [inline index.js:124]: "defining functions" is a little vague, ie a variable binding to a function expression is defining a function and would not be allowed. Also the position as written seems backwards from the decision for functions.

Do you think the following may be clearer?: "However, placing function declarations at the end of a file is common programming practice for readability"

**Review by rajsite — APPROVED** [review]

File: index.js [hunk 2]
```diff
@@ -62,17 +74,68 @@ module.exports = {
...
         /*
             This rule defends against prototype pollution vulnerabilities that exploitable with
             XSS.
         */
         'no-prototype-builtins': 'error',
 
+        /*
+            Higher-order function composition is preferred for iterative loops. For...in statements
+            can be used when performance is a consideration as long as guard-for-in is enabled.
+            For..of statements can be used when break is required for control flow or the iteration
```
Comment (jattasNI) [inline index.js:104]: My nitpicking mind can't stop seeing that we use three dots in `for...in` but two in `for..of`. Can we use three for both?

File: index.js [hunk 3]
```diff
@@ -62,17 +74,68 @@ module.exports = {
...
+        }],
+
         /*
             This rule defends against prototype pollution vulnerabilities that exploitable with
             XSS.
         */
         'no-prototype-builtins': 'error',
 
+        /*
+            Higher-order function composition is preferred for iterative loops. For...in statements
```
Comment (jattasNI) [inline index.js:102]: The phrase "higher-order function composition" might not be immediately understood for some readers (cough...me...cough) so I think adding an example would help. Something like
"Higher-order function composition (array.forEach(...)) is preferred for iterative loops".

Or we could even provide explicit guidance by stealing the guidance from airbnb's readme.
    > Use `map()` / `every()` / `filter()` / `find()` / `findIndex()` / `reduce()` / `some()` / ... to iterate over arrays, and `Object.keys()` / `Object.values()` / `Object.entries()` to produce arrays so you can iterate over objects.

**Review by jattasNI — APPROVED** [review]

**Review by LukasKall — APPROVED** [review]

**Review by mure — APPROVED** [review]

**Comment by rajsite** [comment]
@alejandro5042 @dschwartzni @trstamper can y'll take a look when you have a chance?
P.S. I've submitted a request to have David and Tim added as collaborators in a public repo

**Review by alejandro5042 — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (TrevorKarjanis) [reply index.js:124]: Whoops! Yes, that is fine.

**Pull request merged**